galileo/src/androidTest/java/com/josedlpozo/galileo/realm/RealmSnapshooterTest.kt

```diff
@@ -0,0 +1,69 @@
+package com.josedlpozo.galileo.realm
+
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import com.josedlpozo.galileo.realm.realmbrowser.files.model.RealmFile
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class RealmSnapshooterTest {
+
+    private val realmSnapshooter = RealmSnapshooter()
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getTargetContext())
+        val builder = RealmConfiguration.Builder()
+        with(Realm.getInstance(builder.name("database_developers").deleteRealmIfMigrationNeeded().build())) {
+            executeTransaction {
+                developers.forEach { copyToRealmOrUpdate(it) }
+            }
+            close()
+        }
+        with(Realm.getInstance(builder.name("database_teams").deleteRealmIfMigrationNeeded().build())) {
+            executeTransaction {
+                teams.forEach { copyToRealmOrUpdate(it) }
+            }
+            close()
+        }
+    }
+
+    @Test
+    fun given_realm_databases_when_making_a_snapshoot_then_snapshoot_is_ok() {
+        val snapshoot = realmSnapshooter.shoot(listOf(RealmFile("database_developers", "0", 0L),
+            RealmFile("database_teams", "0", 0L)))
+
+        val expected = "There are 2 Realm databases used by this app\n" +
```

Comment:
@vicfran what do you think about using here this kind of strings?
`
"""There are 2 Realm databases used by this app
            
            Database database_developers has 2 tables
            - DeveloperModel has 2 columns and 6 rows
            | id || name |
            0 fpulido 
            1 jmdelpozo 
            2 vfrancisco 
            3 jaznar 
            4 molmedo 
            5 efau 
            - TeamModel has 2 columns and 0 rows
            | id || name |
            
            
            Database database_teams has 2 tables
            - DeveloperModel has 2 columns and 0 rows
            | id || name |
            - TeamModel has 2 columns and 2 rows
            | id || name |
            0 android 
            1 ios
            
            """.trimIdent()
`

Reply:
I have tested the function and it does not help with our scenario, this is the snapshot generated:

There are 2 Realm databases used by this app

Database database_developers has 2 tables
- DeveloperModel has 2 columns and 6 rows
| id || name |
0 fpulido 
1 jmdelpozo 
2 vfrancisco 
3 jaznar 
4 molmedo 
5 efau 
- TeamModel has 2 columns and 0 rows
| id || name |


Database database_teams has 2 tables
- DeveloperModel has 2 columns and 0 rows
| id || name |
- TeamModel has 2 columns and 2 rows
| id || name |
0 android 
1 ios 

And as fun doc says:

/**
 * Detects a common minimal indent of all the input lines, removes it from every line and also removes the first and the last
 * lines if they are blank (notice difference blank vs empty).
 *
 * Note that blank lines do not affect the detected indent level.
 *
 * In case if there are non-blank lines with no leading whitespace characters (no indent at all) then the
 * common indent is 0, and therefore this function doesn't change the indentation.
 *
 * Doesn't preserve the original line endings.
 *

It removes the last space of each row line, also it removes last lines so expected string is never same as generated :(